front/src/iframe_api.ts

```diff
@@ -1,286 +1,173 @@
-import type { ChatEvent } from "./Api/Events/ChatEvent";
-import { isIframeResponseEventWrapper } from "./Api/Events/IframeEvent";
-import { isUserInputChatEvent, UserInputChatEvent } from "./Api/Events/UserInputChatEvent";
-import { Subject } from "rxjs";
-import { EnterLeaveEvent, isEnterLeaveEvent } from "./Api/Events/EnterLeaveEvent";
-import type { OpenPopupEvent } from "./Api/Events/OpenPopupEvent";
-import { isButtonClickedEvent } from "./Api/Events/ButtonClickedEvent";
-import type { ClosePopupEvent } from "./Api/Events/ClosePopupEvent";
-import type { OpenTabEvent } from "./Api/Events/OpenTabEvent";
-import type { GoToPageEvent } from "./Api/Events/GoToPageEvent";
-import type { OpenCoWebSiteEvent } from "./Api/Events/OpenCoWebSiteEvent";
-import type { PlaySoundEvent } from "./Api/Events/PlaySoundEvent";
-import type { StopSoundEvent } from "./Api/Events/StopSoundEvent";
-import type { LoadSoundEvent } from "./Api/Events/LoadSoundEvent";
-import SoundConfig = Phaser.Types.Sound.SoundConfig;
-import type { LoadPageEvent } from './Api/Events/LoadPageEvent';
+import {registeredCallbacks} from "./Api/iframe/registeredCallbacks";
+import {
+    IframeResponseEvent,
+    IframeResponseEventMap,
+    isIframeResponseEventWrapper,
+    TypedMessageEvent
+} from "./Api/Events/IframeEvent";
+import chat from "./Api/iframe/chat";
+import type {IframeCallback} from './Api/iframe/IframeApiContribution';
+import nav from "./Api/iframe/nav";
+import controls from "./Api/iframe/controls";
+import ui, {ButtonDescriptor, Popup} from "./Api/iframe/ui";
+import sound, {Sound} from "./Api/iframe/sound";
```

Comment:
these imports seems weird - and should also not be in the same file it just gets unreadable

Reply:
Indeed! :+1: